src/main/kotlin/org/oso/core/controllers/HelpProviderController.kt

```diff
@@ -53,10 +54,15 @@ class HelpProviderController
         return ResponseEntity.accepted().build<Unit>()
     }
 
-    private fun getHelpProviderOrFail(id: Long): HelpProvider {
+    private fun getHelpProviderOrFail(id: String): HelpProvider {
         return helpProviderService.findById(id) ?: throw HelpProviderNotFoundException("HelpProvider<$id> not found")
     }
 
+    private fun HelpProviderPushDto.toEntity() = HelpProvider(
+        name = name,
+        keycloakName = SecurityContextHolder.getContext().authentication.name
```

Comment:
We could consider wrapping ``SecurityContextHolder.getContext().authentication.name`` in an own method in a separate service so we can mock it in a test envirionment.